simulator/src/main/java/com/github/benmanes/caffeine/cache/simulator/policy/linked/LinkedPolicy.java

```diff
@@ -66,19 +73,31 @@ public LinkedPolicy(Admission admission, EvictionPolicy policy, Config config) {
     ).collect(toSet());
   }
 
+  @Override
+  public Set<Characteristic> characteristics() {
+    return Sets.immutableEnumSet(WEIGHTED);  
+  }
+
   @Override
   public PolicyStats stats() {
     return policyStats;
   }
-
+  
   @Override
-  public void record(long key) {
+  public void record(AccessEvent event) {
+    final int weight = event.weight();
+    final long key = hashCode(event.key(), weight);
```

Comment:
Why does the key carry the weight?

Reply:
In many traces (including the provided trace of adapt-size) there are few objects with the same key but a different weight (I think the source of it is write operation in between). Those are different objects, so we should handle them as such (other implementations do so as well).